src/plugins/data_source_management/server/saved_objects/data_source_premission_client_wrapper.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import _ from 'lodash';
+import {
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsDeleteOptions,
+  SavedObjectsErrorHelpers,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from '../../../../core/server';
+import { DATA_SOURCE_SAVED_OBJECT_TYPE, ManageableBy } from '../../common';
+import { getWorkspaceState } from '../../../../core/server/utils';
+
+/**
+ * Determine whether the user has the permissions to create, delete, and update data source based on manageableBy/dataSourceAdmin.
+ * DataSourceAdmin user has all permissions.
+ * If manageableBy is all, any user has permissions.
+ * If manageableBy is none, any user has no permissions.
+ * If manageableBy is dashboard_admin, only OSD admin has permissions.
+ */
+export class DataSourcePermissionClientWrapper {
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const { isDashboardAdmin, isDataSourceAdmin } = getWorkspaceState(wrapperOptions.request) || {};
+
+    if (
+      isDataSourceAdmin ||
+      this.manageableBy === ManageableBy.All ||
+      (this.manageableBy === ManageableBy.DashboardAdmin && isDashboardAdmin !== false)
```

Comment:
`isDashboardAdmin !== false` is the judgement here by intension to cover the case that isDashboardAdmin is `undefined`?

Reply:
No. If the isDashboardAdmin is undefined(maybe the workspace/permission control is disabled) or true, we consider  the user is dashboard admin. This is also the reason I used `isDashboardAdmin !== false`

Reply:
So I think it is by intension and could you please add some comment here to elaborate the reason why we have to use `isDashboardAdmin !== false`?

Reply:
Sure, I will add comment to describe why we use `isDashboardAdmin !== false`